doc/3/controllers/document/update-by-query/index.md

```diff
@@ -0,0 +1,76 @@
+---
+code: true
+type: page
+title: updateByQuery
+description: Updates documents matching query
+---
+
+# updateByQuery
+
+Updates documents matching the provided search query.
+
+Kuzzle uses the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
+
+An empty or null query will match all documents in the collection.
+
+<br/>
+
+```java
+  public CompletableFuture<ConcurrentHashMap<String, ArrayList<Object>>> updateByQuery(
+      final String index,
+      final String collection,
+      final ConcurrentHashMap<String, Object> searchQuery,
+      final ConcurrentHashMap<String, Object> changes) throws NotConnectedException, InternalException
+
+  public CompletableFuture<ConcurrentHashMap<String, ArrayList<Object>>> updateByQuery(
+      final String index,
+      final String collection,
+      final ConcurrentHashMap<String, Object> searchQuery,
+      final ConcurrentHashMap<String, Object> changes,
+      final UpdateOptions options) throws NotConnectedException, InternalException
+```
+
+| Argument           | Type                                         | Description     |
+| ------------------ | -------------------------------------------- | --------------- |
+| `index`            | <pre>String</pre>                            | Index name      |
+| `collection`       | <pre>String</pre>                            | Collection name |
+| `searchQuery`      | <pre>ConcurrentHashMap<String, Object></pre> | Query to match  |
+| `changes`          | <pre>ConcurrentHashMap<String, Object></pre> | Partial changes to apply to the documents |
+| `options`          | <pre>UpdateOptions</pre><br>(`null`)         | Optional parameters               |
+
+---
+
+### options
+
+A [UpdateOptions](/sdk/java/3/core-classes/update-options) object.
+
+The following options can be set:
+
+| Arguments          | Type                                         | Description                       |
+| ------------------ | -------------------------------------------- | --------------------------------- |
+| `waitForRefresh`   | <pre>Boolean</pre>                           | If set to `true`, Kuzzle will wait for the persistence layer to finish indexing|
+| `retryOnConflict`  | <pre>Integer</pre>                           | The number of times the database layer should retry in case of version conflict |
```

Comment:
This option is not part of the `updateByQuery` API

Reply:
https://github.com/kuzzleio/kuzzle/blob/2-dev/lib/services/elasticsearch.js#L730
https://github.com/kuzzleio/kuzzle/blob/2-dev/lib/api/controllers/document.js#L480
?